app/components/Settings/ConnectButton/ConnectButton.react.js

```diff
@@ -20,30 +20,42 @@ export default class ConnectButton extends Component {
         let error = null;
 
         if (connectRequest.status >= 400 || saveConnectionsRequest.status >= 400) {
-           buttonText = 'Connect';
-           buttonClick = connect;
-           // TODO - Try out locking the home folder and verifying this.
-           let errorMessage = 'Hm... had trouble connecting.';
-           if (connectRequest.content && connectRequest.content.error) {
-               errorMessage = connectRequest.content.error;
-           } else if (saveConnectionsRequest.content && saveConnectionsRequest.content.error && saveConnectionsRequest.content.error.message) {
-               errorMessage = saveConnectionsRequest.content.error.message;
-           }
-           error = <div className={styles.errorMessage}>{errorMessage}</div>;
-       } else if (!connectRequest.status) {
             buttonText = 'Connect';
             buttonClick = connect;
-        } else if (connectRequest.status === 'loading') {
+            // TODO - Try out locking the home folder and verifying this.
+            let errorMessage = 'Hm... had trouble connecting.';
+            if (
+               has('conent', connectRequest) &&
+               has('error', connectRequest.content)
```

Comment:
Except I guess if the `fetch` failed because of a network issue. I'd try manually testing connecting with like your wifi turned off and see what is in the `connectRequest` object. I forget how that's handled.